### PR TITLE
Update Netcetera SDK and replace deprecated methods

### PR DESCRIPTION
**Pull request by AnasNaouchi**
- Update Netcetera Library
- Replace License key with api key
- Update UI customization code to match with the new standard

### CODE REVIEW

File: sdk/build.gradle [hunk 1]
```diff
@@ -19,6 +19,7 @@ android {
         buildConfigField "int", "VERSION_CODE", "$omise_sdk_code_version"
         buildConfigField "String", "VERSION_NAME", "\"$omise_sdk_version\""
         buildConfigField "String", "NETCETERA_LICENSE_KEY", "\"${System.getenv('NETCETERA_LICENSE_KEY')}\""
```
Comment (vault087) [inline sdk/build.gradle:21]: It seems we are no longer need NETCETERA_LICENSE_KEY, right?

**Review by danfowler — APPROVED** [review]

**Comment by danfowler** [comment]
SonarCloud is reporting a deprecation warning, and it looks like we are using the legacy initialization https://3dss.netcetera.com/3dssdk/doc/2.11.0/android-sdk-api

File: sdk/build.gradle (hunk 1 above)
Comment (AnasNaouchi) [reply sdk/build.gradle:21]: yes will remove

File: app/src/kotlin/java/co/omise/android/example/CheckoutActivity.kt [hunk 2]
```diff
@@ -41,7 +43,7 @@ class CheckoutActivity : AppCompatActivity() {
     companion object {
 
         private const val TAG = "CheckoutActivity"
-        private const val PUBLIC_KEY = "[PUBLIC_KEY]"
+        private const val PUBLIC_KEY = "pkey_5xjdtti64j5cwzxa9xo"
```
Comment (aashishgurung) [inline app/src/kotlin/java/co/omise/android/example/CheckoutActivity.kt:46]: Is it required to include a valid `pkey`?

File: sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt [hunk 3]
```diff
@@ -188,7 +188,7 @@ class AuthorizingPaymentActivity : AppCompatActivity() {
     }
 
     private fun setupActionBarTitle() {
-        supportActionBar?.title = uiCustomization.uiCustomization.toolbarCustomization?.headerText
+        supportActionBar?.title = uiCustomization.uiCustomizationMap[com.netcetera.threeds.sdk.api.ui.logic.UiCustomization.UiCustomizationType.DEFAULT]?.toolbarCustomization?.headerText
```
Comment (aashishgurung) [inline sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt:191]: Why does the key of `uiCustomizationMap` needs to that long?

File: app/src/kotlin/java/co/omise/android/example/CheckoutActivity.kt (hunk 2 above)
Comment (AnasNaouchi) [reply app/src/kotlin/java/co/omise/android/example/CheckoutActivity.kt:46]: oh yes sorry i should remove

File: sdk/src/main/java/co/omise/android/ThreeDS2ServiceWrapper.kt [hunk 4]
```diff
@@ -37,12 +37,20 @@ internal class ThreeDS2ServiceWrapper(
...
                 .build()
             val configParameters = ConfigurationBuilder()
-                .license(BuildConfig.NETCETERA_LICENSE_KEY)
+                .apiKey(BuildConfig.NETCETERA_API_KEY)
                 .configureScheme(schemeConfig)
                 .build()
             val locale = getLocale()
-            threeDS2Service.initialize(context, configParameters, locale, uiCustomization)
-            continuation.resume(Result.success(Unit))
+            threeDS2Service.initialize(context, configParameters, locale, uiCustomizationMap,object :
```
Comment (danfowler) [inline sdk/src/main/java/co/omise/android/ThreeDS2ServiceWrapper.kt:44]:
```suggestion
            threeDS2Service.initialize(context, configParameters, locale, uiCustomizationMap, object :
```

Comment (danfowler) [reply sdk/src/main/java/co/omise/android/ThreeDS2ServiceWrapper.kt:44]: @AnasNaouchi please make a ticket for implementing and enforcing some kind of linter for these kinds of things?

File: sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt (hunk 3 above)
Comment (danfowler) [reply sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt]: What about

```
import com.netcetera.threeds.sdk.api.ui.logic.UiCustomization.UiCustomizationType.DEFAULT as defaultUICustomization
```

or something like it?

**Review by aashishgurung — APPROVED** [review]

**Review by danfowler — APPROVED** [review]

**Pull request merged**